tests/unit/clients/workflow_test.py

```diff
@@ -54,3 +77,239 @@ def test_terminate_workflow(workflow_config, workflow_client, boto_client):
+        typeFilter=dict(
+            name='test',
+            version='test',
+        ),
+    )
+
+    workflow_client.count_closed_workflow_executions(
+        oldest_close_date=oldest_close_date,
+        workflow_name='test',
+        version='test',
+    )
+    _assert_count_closed_workflow_executions_called_with(
+        boto_client,
+        workflow_config,
+        closeTimeFilter=dict(oldestDate=oldest_close_date),
+        typeFilter=dict(
+            name='test',
+            version='test',
+        ),
+    )
+
+    workflow_client.count_closed_workflow_executions(
+        oldest_start_date=oldest_start_date,
+        tag='test',
+    )
+    _assert_count_closed_workflow_executions_called_with(
+        boto_client,
+        workflow_config,
+        startTimeFilter=dict(oldestDate=oldest_start_date),
+        tagFilter=dict(tag='test'),
+    )
+
+    workflow_client.count_closed_workflow_executions(
+        oldest_close_date=oldest_close_date,
+        tag='test',
+    )
+    _assert_count_closed_workflow_executions_called_with(
+        boto_client,
+        workflow_config,
+        closeTimeFilter=dict(oldestDate=oldest_close_date),
+        tagFilter=dict(tag='test'),
+    )
+
+    workflow_client.count_closed_workflow_executions(
+        oldest_start_date=oldest_start_date,
+        workflow_id='test',
+    )
+    _assert_count_closed_workflow_executions_called_with(
+        boto_client,
+        workflow_config,
+        startTimeFilter=dict(oldestDate=oldest_start_date),
+        executionFilter=dict(workflowId='test'),
+    )
+
+    workflow_client.count_closed_workflow_executions(
+        oldest_close_date=oldest_close_date,
+        workflow_id='test',
+    )
+    _assert_count_closed_workflow_executions_called_with(
+        boto_client,
+        workflow_config,
+        closeTimeFilter=dict(oldestDate=oldest_close_date),
+        executionFilter=dict(workflowId='test'),
+    )
+
+    workflow_client.count_closed_workflow_executions(
+        oldest_start_date=oldest_start_date,
+        close_status='test',
+    )
+    _assert_count_closed_workflow_executions_called_with(
+        boto_client,
+        workflow_config,
+        startTimeFilter=dict(oldestDate=oldest_start_date),
+        closeStatusFilter=dict(status='test'),
+    )
+
+    workflow_client.count_closed_workflow_executions(
+        oldest_close_date=oldest_close_date,
+        close_status='test',
+    )
+    _assert_count_closed_workflow_executions_called_with(
+        boto_client,
+        workflow_config,
+        closeTimeFilter=dict(oldestDate=oldest_close_date),
+        closeStatusFilter=dict(status='test'),
+    )
+
+
+def _assert_count_open_workflow_executions_called_with(boto_client, workflow_config, **kwargs):
+    _assert_boto_method_called_with(boto_client.count_open_workflow_executions, workflow_config.domain, **kwargs)
+
+
+def _assert_count_closed_workflow_executions_called_with(boto_client, workflow_config, **kwargs):
```

Comment:
Let's remove these calls since the level of indirection isn't necessary and would make things more explicitly when asserting calls.